urllib3/util/url.py

```diff
@@ -158,6 +158,11 @@ def parse_url(url):
     if '://' in url:
         scheme, url = url.split('://', 1)
 
+    # Auth
+    if '@' in url:
+        # Last '@' denotes end of auth part
```

Comment:
The "auth" part is not one of the actual components of a URI, however. The components are:

- scheme
- authority (not auth by way urllib3 refers to auth)
- path
- query
- fragment

An authority can be parsed into credentials, hostname, and port, but the parsing of those three sub-components should be done after parsing out the authority.